actionpack/lib/action_dispatch.rb

```diff
@@ -52,9 +52,10 @@ class MissingController < NameError
     autoload_under "http" do
       autoload :ContentSecurityPolicy
       autoload :PermissionsPolicy
-      autoload :Request
       autoload :Response
+      autoload :URI
     end
+    autoload :Request
```

Comment:
Why was this moved?

Reply:
The class is called ActionDispatch::Request and not ActionDispatch::Http::Request.
Plus, the whole http folder is a bit chaotic, this is a small first cleanup step.

Reply:
I got that, but I think we should not try to mix these changes as it creates unnecessary confusion. What if this introduces a regression? We have to revert this whole PR? It's better to do unrelated refactors separately IMO.

Reply:
It is only the location of the file that got corrected, nothing changed that is of concern. You can still patch ActionDispatch::Reqest as you could before, not change at all.

Reply:
> It is only the location of the file that got corrected, nothing changed that is of concern.

While the Ruby code may be equivalent, this makes git operations more difficult. As zzak mentioned, what happens if this PR needs to be reverted (or what if a change to the old file location needs to be reverted)? If bug fixes are made to the new file location, backporting them becomes more work than just cherry picking the commit. 

In general, purely cosmetic changes are rejected https://github.com/rails/rails/pull/13771#issuecomment-32746700 I'm not sure there's a strong enough reason to change this yet

Reply:
I give up. You wanna argue against fixing small details, go ahead. I reverted my fix.Lets keep it a bit more chaotic as it is. And yes, git blame gets a bit more complicated, but it is just a file in the wrong place that makes it harder to understand as a developer.

Reply:
sure.